R/download_climate.R

```diff
@@ -0,0 +1,386 @@
+        if ("date" %in% colnames(climate_data)) {
+          climate_data <- merge(climate_data,
+            stations_data,
+            by.x = "date",
+            by.y = "date",
+            all.x = TRUE
+          )
+        } else {
+          climate_data <- stations_data
+        }
+      }
+      names(climate_data) <- c("date", tags)
+    } else {
+      climate_data <- list()
+      for (tag in tags) {
+        stations_data <- retrieve_stations_data(
+          stations = stations,
+          start_date = start_date,
+          end_date = end_date,
+          frequency = frequency,
+          tag = tag,
+          aggregate = aggregate
+        )
+        climate_data[[tag]] <- stations_data
+      }
+    }
+  }
+  return(climate_data)
+}
+
+
+#' Retrieve data from named stations for a specific tag
+#'
+#' @description
+#' Retrieve climate data from a list of stations under the same tag.This data is
+#' retrieved from local meteorological stations provided by IDEAM
+#'
+#' @param stations numeric vector containing the stations' codes
+#' @param start_date character with the first date to consult in the format
+#' \code{"YYYY-MM-DD"}
+#' @param end_date character with the last date to consult in the format
+#' \code{"YYYY-MM-DD"} (Last available date is \code{"2023-05-31"})
+#' @param frequency character with the aggregation frequency. (\code{"day"},
+#' \code{"week"}, \code{"month"} or \code{"year"})
+#' @param tag unique character containing climate tags to consult
+#' @param aggregate logical for data aggregation, if \code{FALSE}, returns the
+#' data from each individual station in the area. (Default = \code{TRUE})
+#'
+#' @importFrom rlang .data
+#' @importFrom magrittr %>%
+#'
+#' @return \code{data.frame} with observations from the requested stations
+#'
+#' @keywords internal
+retrieve_stations_data <- function(stations, start_date, end_date,
+                                   frequency, tag,
+                                   aggregate = TRUE) {
+  checkmate::assert_vector(stations)
+  ideam_tags <- c(
+    "TSSM_CON", "THSM_CON", "TMN_CON", "TMX_CON",
+    "TSTG_CON", "HR_CAL", "HRHG_CON", "TV_CAL",
+    "TPR_CAL", "PTPM_CON", "PTPG_CON", "EVTE_CON", "FA_CON",
+    "NB_CON", "RCAM_CON", "BSHG_CON", "VVAG_CON",
+    "DVAG_CON", "VVMXAG_CON", "DVMXAG_CON"
+  )
+  checkmate::assert_choice(tag, ideam_tags)
+  checkmate::assert_logical(aggregate)
+
+  path_data <- retrieve_path("IDEAM_CLIMATE_2023_MAY")
+  path_stations <- paste0(tag, "@", stations, ".data")
+  date_range <- seq(as.Date(start_date),
+    as.Date(end_date),
+    by = frequency
+  )
+  floor_dates <- lubridate::floor_date(date_range,
+    unit = frequency
+  )
+  stations_data <- data.frame(date = floor_dates)
+  for (i in seq_along(path_stations)) {
+    # nolint start: nonportable_path_linter
+    dataset_path <- file.path(path_data, path_stations[i])
+    # nolint end
+    station <- data.frame(NA)
+    tryCatch(
```

Comment:
As stated lower, this was an immediate solve. Climate data is stored in multiple files, and sometimes, given the quality filters of the source, data is not available and the request we perform can return an error (since the data is not found). This was better considered in #67, where the TryCatch was removed and we included a condition based in the http response.